packages/react-reconciler/src/ReactFiberDevToolsHook.js

```diff
@@ -123,19 +65,64 @@ export function injectInternals(internals: Object): boolean {
 }
 
 export function onScheduleRoot(root: FiberRoot, children: ReactNodeList) {
-  if (typeof onScheduleFiberRoot === 'function') {
-    onScheduleFiberRoot(root, children);
+  if (__DEV__) {
+    if (
+      injectedHook &&
+      typeof injectedHook.onScheduleFiberRoot === 'function'
+    ) {
```

Comment:
Could sent pointers to e.g. `onScheduleFiberRoot` directly so we wouldn't have to do the property access

```js
if (typeof onScheduleFiberRoot === 'function') {
  onScheduleFiberRoot(rendererID, root, chilren);
}
```

Reply:
I think Seb is saying that property gets mutated by Fast Refresh. (I don't actually remember if it does but I guess?)